tests/Application/ConfigurationTest.php

```diff
@@ -257,7 +257,7 @@ public static function configurationWithoutTargetProvider(): iterable
                         ],
                     ],
                 ],
-                'if' => "obj.owner !== null && (obj.owner.firstName === 'John')",
+                'if' => 'obj.owner !== null && ((obj.owner.firstName === "John"))',
```

Comment:
when converting some types of nodes to string, they are wrapped in parentheses so we end up with double parentheses.
not sure if we should handle this or just leave it be.